src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let copyState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        copyState = { ...copyState, ...action.extraData };
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete copyState[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        copyState = {};
+        break;
+      }
+    }
```

Comment:
Consider adding a default case in the switch statement to handle any unexpected action types. This would improve error handling and ensure that your function behaves predictably even when encountering unknown actions.